src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1530,6 +1530,12 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
             }
 
             return null;
+
+            static bool ignoreReceiver(BoundExpression receiver, MethodSymbol method)
+            {
+                // ignore the implicit `this` receiver on a static method
+                return method.IsStatic && receiver is { Kind: BoundKind.ThisReference, WasCompilerGenerated: true };
```

Comment:
That might be too broad. AnalyzeDataFlow should still see the invalid receiver as being read for extract method to work as intended.

Reply:
> AnalyzeDataFlow should still see the invalid receiver as being read for extract method to work as intended.

I am not sure if I agree with that. Clearly, the receiver for static methods (excluding extension scenario) is not going to be evaluated, so I think we have a flexibility in terms of how flow analysis should work in the error scenario.
The current implementation is incorrect in my opinion. Primarily because the PR description provides no clarity about what is our philosophy around visiting and not visiting receivers. Also, it is not clear to me why do we want to handle receivers differently by comparison to BoundCall. Perhaps the behavior is the same, but it is not obvious from the code.

Reply:
In general, I do not see us actually making sure that region analysis will work for any misplaced expression. For example, there are ```BoundLocalDeclaration.ArgumentsOpt``` and it looks like flow analysis never visits them. Therefore, it is not clear to me why we should visit erroneous (misplaced) receiver. BTW, the issue that we are trying to fix is about failure for a valid scenario.